### DISCUSSION TITLE
When launching 2 cypress-cloud run command for a same pipeline by specifying a tag, the second tagged run is not run.

### DISCUSSION THREAD

**Discussion by samixchoumi** [currents-dev/cypress-currents-reporter#148] [Q&A]
### Before opening, please confirm:

- [X] I have searched for [duplicate or closed issues](https://github.com/currents-dev/cypress-cloud/issues) and [discussions](https://github.com/currents-dev/cypress-cloud/discussions).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I acknowledge that I will attach a **full debug log**, otherwise the issue will be closed with no response.

### Environment information

<details>

```

System:
    OS: macOS 13.4.1
    CPU: (8) arm64 Apple M1 Pro
    Memory: 59.88 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.12.1 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
    Firefox: 113.0.2
    Safari: 16.5.1
  npmPackages:
    @badeball/cypress-cucumber-preprocessor: ^18.0.1 => 18.0.1 
    @bahmutov/cypress-esbuild-preprocessor: ^2.2.0 => 2.2.0 
    @cypress/angular:  0.0.0-development 
    @cypress/mount-utils:  0.0.0-development 
    @cypress/react:  0.0.0-development 
    @cypress/react18:  0.0.0-development 
    @cypress/svelte:  0.0.0-development 
    @cypress/vue:  0.0.0-development 
    @cypress/vue2:  0.0.0-development 
    @faker-js/faker: ^8.0.2 => 8.0.2 
    @shelex/cypress-allure-plugin: ^2.40.0 => 2.40.0 
    cy2: ^4.0.6 => 4.0.6 (3.4.3)
    cypress: ^12.14.0 => 12.15.0 
    cypress-cloud: ^1.7.4 => 1.7.4 
    cypress-multi-reporters: ^1.6.3 => 1.6.3 
    cypress-real-events: ^1.8.1 => 1.8.1 
    cypress-recurse: ^1.35.2 => 1.35.2 
    cypress-slack-reporter: ^1.5.3 => 1.5.3 
    eslint: ^8.43.0 => 8.43.0 
    eslint-plugin-cypress: ^2.13.3 => 2.13.3 
    mochawesome: ^7.1.3 => 7.1.3 
    mochawesome-merge: ^4.3.0 => 4.3.0 
    mochawesome-report-generator: ^6.2.0 => 6.2.0 
    typescript: ^5.1.3 => 5.1.3 
  npmGlobalPackages:
    allure-commandline: 2.22.4
    corepack: 0.14.2
    npm: 9.6.4

```

</details>

### Describe the bug

In my GitLab CI, I run 2 commands : 
- `npx cypress-cloud run --record --key aaa --parallel --ci-build-id $CI_PIPELINE_ID --env allure=true,configFile=preprod,TAGS=\"not @ignore and not @teststeardown\" --browser chrome`
- `npx cypress-cloud run --ci-build-id $CI_PIPELINE_ID --env allure=false,configFile=preprod,TAGS=\"@teststeardown\" --browser chrome`

The problem is that second command is skipped because for `cypress-cloud`, the tagged test with `@teststeardown` are already "run" (or skipped) on the previous call, so they are not launched but I need them to be specifically launched AFTER the first run.

If I change the order, this time, the `@teststeardown` tags should be launch but the step is just skipped.

### Expected behavior

When a test is tagged to be not run, just don't put it to ignore case (except if tagged ignore).

### Command and Setup

I use sorry-cypress provider. 
The commands are shown before.

### Full log and debug output

<details>

```

No error logs. 

```

Just the stage is skipped : 
![image](https://github.com/currents-dev/cypress-cloud/assets/22171928/0c7d6cb5-cd0b-4501-b9af-269ff8031545)

</details>

**Comment by agoldis** [comment]
@samixchoumi Please refer to https://currents.dev/readme/guides/cypress-ci-build-id guide. Also, please consider using `--group` to have separate groups for the same run

**Reply by samixchoumi** [reply]
Hello @agoldis and thanks you for your answer, where can I found the documentation about `--group` ? Because for the moment I changed the use of `cypress-cloud` usage for the run of `@teststeardown` to `cypress`. So it's not skipped.

**Reply by agoldis** [reply]
On cypress's official documentation
or just try adding `--group xxx` to the CLI command

**Reply by samixchoumi** [reply]
I will look at this solution (doc here : https://docs.cypress.io/guides/guides/command-line#cypress-run-group-lt-name-gt). Hope it will work

**Reply by samixchoumi** [reply]
So it doesn't work using `--group` option, the better way is to use `cypress run` for one step (that's not mandatory to appear on the dashboard) and then `cypress-cloud` run for the mandatory step.
